metrics/default_metrics/luajit/init.lua

```diff
@@ -0,0 +1,75 @@
+local metrics = require('metrics')
+
+local has_mics_module, misc = pcall(require, 'misc')
+
+local gauges_storage = {}
+
+local LJ_PREFIX = 'lj_'
+
+local function prefix_name(name)
+    return LJ_PREFIX .. name
+end
+
+local function set_gauge(name, description, value, labels)
+    if (gauges_storage[name] == nil) then
+        gauges_storage[name] = metrics.gauge(prefix_name(name), description)
+    end
+
+    gauges_storage[name]:set(value, labels or {})
+end
+
+local function update()
+    local lj_metrics = misc.getmetrics()
+    set_gauge('gc_freed', 'Total amount of freed memory',
+        lj_metrics.gc_freed)
+    set_gauge('strhash_hit', 'Number of strings being interned',
+        lj_metrics.strhash_hit)
+    set_gauge('gc_steps_atomic', 'Count of incremental GC steps (atomic state)',
+        lj_metrics.gc_steps_atomic)
+    set_gauge('strhash_miss', 'Total number of strings allocations during the platform lifetime',
```

Comment:
This one and probably some others should be counters, not gauges, IMO

Reply:
You are right but counters support only "inc" operation. I used gauges here because them allow to "set" value.

Reply:
both points are valid. it must be counter and there must be no `set` method on counter. 

[`Counter`.`Set`](https://github.com/prometheus/client_golang/blob/434a8ed85d5c61523c49d9ebb3fc49379ddef998/prometheus/counter.go#L39) has been part of prometheus api, until it has been removed in favour of [`NewConstMetric`](https://github.com/prometheus/client_golang/blob/e6ea98bdda971dabab09e8d93a8d65885460fd1e/prometheus/value.go#L80) approach, [example usage](https://www.robustperception.io/setting-a-prometheus-counter)

We should distinguish two use cases:

- instrumenting applications where we are in charge of calling `inc` on coounter

- exposing already collected metrics, like luajit

First case is clear - only `inc` is required (allowed).

Second case is different. We have already aggregated value which we need to export in appropriate type (read [this discussion in prometheus-net client](https://github.com/prometheus-net/prometheus-net/issues/32) for more details). This is the case where `NewConstMetric` workflow is perfectly valid. Until you are using golang.

We should use appropriate metric types (in this case Counter). 
I think there are several possible ways to deal with this:

- use `reset`/`inc` workflow (which is valid for prometheus counter) - `reset(0)` then `inc(current_value)`

- implement new Counter function to set current_value which will result in gauge like behaviour

- adhoc counter extension - add set method to Counter objects, e.g. implement something like
    ```lua
     function create_special_counter()
         local c = new Couner()
         c.set = function() ... end
         return c
      end
    ```

Reply:
I think, extend counter is the best way: user will be understand that he use *extension* (some extra options)
 and won't be confused when to use `inc` and when `set` if default counter will have both methods.

Reply:
I did in this way - https://github.com/tarantool/metrics/pull/128/commits/bfed2a8bef2ab74d33b485a52b6875ddd1a747a8